animation_nodes/nodes/spline/splines_from_edges_utils.pyx

```diff
@@ -0,0 +1,120 @@
+from libc.string cimport memset
+from cpython.mem cimport PyMem_Malloc, PyMem_Realloc, PyMem_Free
+from ... data_structures cimport (
+    Mesh, Vector3DList, EdgeIndicesList, PolySpline,
+    VirtualDoubleList, FloatList
+)
+
+def splinesFromBranches(Vector3DList vertices, EdgeIndicesList edges, VirtualDoubleList radii):
+    cdef int i, j
+    cdef int edgesAmount = edges.length
+    cdef int verticesAmount = vertices.length
+
+    # Compute how many neighbour each vertex have.
+    cdef int *neighboursAmounts = <int*>PyMem_Malloc(verticesAmount * sizeof(int))
+    memset(neighboursAmounts, 0, verticesAmount * sizeof(int))
+    for i in range(edgesAmount):
+        neighboursAmounts[edges.data[i].v1] += 1
+        neighboursAmounts[edges.data[i].v2] += 1
+
+    # Compute the start index of each group of neighbours of each vertex.
+    cdef int *neighboursStarts = <int*>PyMem_Malloc(verticesAmount * sizeof(int))
+    cdef int start = 0
+    for i in range(verticesAmount):
+        neighboursStarts[i] = start
+        start += neighboursAmounts[i]
+
+    # Keep track of how many index is in each group of neighbours at each iteration.
+    cdef int *filledSpaces = <int*>PyMem_Malloc(verticesAmount * sizeof(int))
```

Comment:
I still think you should use an `IntegerList` instead of allocating it yourself.
`IntegerList.fromValue(0, length = verticesAmount)`

Reply:
OK, so, in general, we shouldn't use malloc if there is a list alternative.

Reply:
Right, by using lists the code is safer, we can't forget to free and it is cleaner.

Reply:
Everything now uses lists.